### DISCUSSION TITLE
Ran into issue running `brew upgrade` on mojave, packages broken, now what?

### DISCUSSION THREAD

**Discussion by acagastya** [Homebrew/discussions#2377] [Everyday usage]
Hi, I was trying to use one of the formulæs I'd installed using brew on mojave, and looked like it had an update.  I ran `brew update` and then `brew upgrade` on mojave.  Turns out support for mojave was dropped.

This comes in the terminal:
```
Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 11.3.1.

==> ./bootstrap --prefix=/usr/local/Cellar/cmake/3.21.4 --no-system-libs --parallel=8 --datadir=/share/cmake --docdir=/share/doc/c
==> make
```
and goes on and heats up the laptop (or when trying to reinstall, errors out and has non-ZERO exit; it also mentions, don't report the issue).

I tried reinstalling brew, but to no luck.  All the important binaries are now gone or broken and I am in a paralysed situation.  What can I possibly do now, to at least get the binaries I had, back?

**Comment by SMillerDev** [comment]
Brew does not delete anything that is still in use so all the binaries are still there. The problem is that Mojave is not supported by Apple or Homebrew anymore, so you'll have to install new packages from source or update your system.

**Reply by acagastya** [reply]
The packages were corrupted during the `./bootstrap --prefix=/usr/local/Cellar/cmake/3.21.4 --no-system-libs --parallel=8 --datadir=/share/cmake --docdir=/share/doc/c` phase and were deleted while reinstalling brew.

No way to install automatically, right?

**Reply by SMillerDev** [reply]
If you reinstall brew it will remove all existing packages, there is nothing we can do about this. Your only options are updating the OS to a newer version, or debugging and fixing the build issues.

**Reply by acagastya** [reply]
I can't update the OS.  So I think seeing the formula to build those is the only option.
